src/components/TrialBanner.js

```diff
@@ -172,8 +172,11 @@ export const TrialBanner = _.flow(
           onClick: async () => {
             try {
               await User.finalizeTrial()
+              await refreshTerraProfile()
             } catch (error) {
               reportError('Error finalizing trial', error)
+            } finally {
+              this.setState({ finalizeTrial: false })
```

Comment:
Could also consider doing this as the last thing in the `try` block. It depends on whether or not you want to dismiss the modal when there's an error. I might favor moving this so that it doesn't dismiss but it's probably fine either way.

Team, do we have a guideline for how Terra should behave in cases like that?

Reply:
I elected to do it this way because the error banner would appear behind the modal if there was an error. 

Not sure if we have a guideline, but I agree we should so that it's consistent!